### DISCUSSION TITLE
Time to look at getLocalPlayerHospital?

### DISCUSSION THREAD

**Discussion by lewri** [CorsixTH/CorsixTH#1771] [General Diagnostics]
Inside the world code we have ``World:getLocalPlayerHospital()`` that returns the player's hospital.
In recent PRs we have started work to consider the multiplayer/ai aspect of the game. And as the function above points out it has been  written with ``hospitals[1]`` being set as the player hospital.

https://github.com/CorsixTH/CorsixTH/blob/541540bc9f95719269d9129196f53493ef3e988f/CorsixTH/Lua/world.lua#L431-L437

Is it time to address this TODO do we think? From my limited experience and memory I think there were some limited discussion about this -- with a potential solution maintaining ``hospitals[1]`` as the local player for multiplayer (i.e. everyone has their own ``hospitals[1]``. But I'm sure there are other solutions here. Trouble is I couldn't tell you what one is best.

**Comment by Alberth289346** [comment]
> But I'm sure there are other solutions here.

I would expect that too.

> Trouble is I couldn't tell you what one is best.

I think nobody knows yet.

Do you want to find out?

**Comment by Alberth289346** [comment]
As a side-note, please don't make the assumption that local player hospital is always also `ui.hospital`. I'd like to keep the option of viewing an opponent hospital. Not just the visual view, but also the financial details, the case book, etc etc.
